### PR TITLE
Pic2card designer Better Error messages and Readme updates

### PR DESCRIPTION
**Pull request by Davakumar**
Updated the pic2card details in  adaptive cards designer README file and handled error scenario on service failure.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/4795)

Related issue: https://github.com/microsoft/AdaptiveCards/issues/4708

### CODE REVIEW

File: source/nodejs/adaptivecards-designer/src/open-image-dialog.ts [hunk 1]
```diff
@@ -273,24 +273,33 @@ export class OpenImageDialog extends Dialog {
...
-            for (let template of res.templates) {
-                let sampleImage = new ImageItem(template);
-                sampleImage.onClick = (selectedImage: string) => {
-                    this.renderImage(selectedImage);
-                };
-                imageContainer.appendChild(sampleImage.render());
+            if (res) {
+                const imageTitle = document.createElement("div");
+                imageTitle.innerText = "Choose from our Sample Images";
+                imageTitle.style.marginLeft = "15px";
```
Comment (BalajiR) [inline source/nodejs/adaptivecards-designer/src/open-image-dialog.ts:279]: Can we move this margin styling into the class `acd-image-title` ?

File: source/nodejs/adaptivecards-designer/src/open-image-dialog.ts [hunk 2]
```diff
@@ -273,24 +273,33 @@ export class OpenImageDialog extends Dialog {
...
+                for (let template of res.templates) {
+                    let sampleImage = new ImageItem(template);
+                    sampleImage.onClick = (selectedImage: string) => {
+                        this.renderImage(selectedImage);
+                    };
+                    imageContainer.appendChild(sampleImage.render());
+                }
+                sampleImageTemplate.appendChild(imageContainer);
+            } else {
+                const imageTitle = document.createElement("div");
```
Comment (BalajiR) [inline source/nodejs/adaptivecards-designer/src/open-image-dialog.ts:296]: Shall we rename the constant as `errorElement` or `errorTitle` ?

**Review by matthidinger — APPROVED** [review]

**Review by BalajiR — CHANGES_REQUESTED** [review]

File: source/nodejs/adaptivecards-designer/src/open-image-dialog.ts (hunk 2 above)
Comment (Davakumar) [reply source/nodejs/adaptivecards-designer/src/open-image-dialog.ts]: Updated the variable name. Addressed the review comments.

File: source/nodejs/adaptivecards-designer/src/open-image-dialog.ts (hunk 1 above)
Comment (Davakumar) [reply source/nodejs/adaptivecards-designer/src/open-image-dialog.ts]: Moved the style to css file. Addressed the review comments.

File: source/nodejs/adaptivecards-designer/README.md [hunk 3]
```diff
@@ -260,4 +260,17 @@ designer.sampleData = {
 	name: "John Doe",
 	phone: "123-123-1234"
 };
-```
\ No newline at end of file
+```
+
+## Hosting Pic2Card Service
+
+Pic2Card is a ML based service which converts GUI image into adaptive card payload JSON.
```
[the remark this answers is not in the document]
Comment (Davakumar) [reply source/nodejs/adaptivecards-designer/README.md]: Addressed the review comments.

**Review by BalajiR — APPROVED** [review]
LGTM

File: source/nodejs/adaptivecards-designer/README.md [hunk 4]
```diff
@@ -260,4 +260,17 @@ designer.sampleData = {
...
+
+Pic2Card is a ML based service which converts GUI image into adaptive card payload JSON. 
+
+To setup the pic2card ML service follow the instructions given in the below link.
+
+[pic2card service](
+https://github.com/microsoft/AdaptiveCards/tree/main/source/pic2card#run-the-pic2card-service-in-docker-container
+)
+
+Once your pic2card ML service is up and running, update the host configuration(PIC_TO_CARD_PREDICTION_API)
```
[the remark this answers is not in the document]
Comment (Davakumar) [reply source/nodejs/adaptivecards-designer/README.md:277]: @matthidinger We have added a new configuration in ACDesigner, that helps npm user to configure the pic2card service via API. .env helps for adaptivecards-site release purpose. We have tested both scenarios and its working as expected.

**Pull request merged**